### PR TITLE
refactor: add / use 'Client._get_resource' method

### PR DESCRIPTION
**Pull request by tseaver**
Use an explicit helper client method for `GET` requests, rather than manipulating client's private `_connection.api_request`.  As a benefit, tests get *way* clearer.

Toward #38

~~Based on top of the branch from #430.  I will rebase when that PR merges.~~

### CODE REVIEW

**Comment by tseaver** [comment]
I rebased this PR on top of #431, in order to avoid having the python 2.7 system test install fail.  I will rebase to master once #431 is merged.

**Comment by tseaver** [comment]
Flaky systest reported in #435.  Rerunning tests.

File: google/cloud/storage/blob.py [hunk 1]
```diff
@@ -2829,13 +2828,12 @@ def get_iam_policy(
         if requested_policy_version is not None:
             query_params["optionsRequestedPolicyVersion"] = requested_policy_version
 
-        info = client._connection.api_request(
-            method="GET",
-            path="%s/iam" % (self.path,),
+        info = client._get_resource(
+            "%s/iam" % (self.path,),
```
Comment (chrisrossi) [inline google/cloud/storage/blob.py:2832]: This is more of just a style thing, but in modern Python, should we be preferring `str.format` or `f` strings if Python 2 support has been dropped?

**Review by chrisrossi — APPROVED** [review]
LGTM!

File: google/cloud/storage/blob.py (hunk 1 above)
Comment (tseaver) [reply google/cloud/storage/blob.py:2832]: Once we drop Python2, I would definitely prefer f-strings.

**Pull request merged**